azure-iot-device/tests/common/pipeline/pipeline_stage_test.py

```diff
@@ -289,3 +215,280 @@ def test_enters_pipeline(self, mocker, stage, method_name, fake_non_pipeline_thr
                 assert pipeline_executor.submit.call_count == 1
 
     setattr(module, "Test{}PipelineThreading".format(cls.__name__), LocalTestObject)
+
+
+def _add_pipeline_flow_tests(cls, module):
+    class PipelineFlowTestBase(object):
```

Comment:
>PipelineFlowTestBase [](start = 10, length = 20)

I assume this base class is  here because this is a WIP.  otherwise it doesn't make much sense.

Reply:
I use it to provide the stage fixture. If you have a better idea I'm interested to hear it

---
In reply to: [338066181](https://github.com/Azure/azure-iot-sdk-python/pull/369#discussion_r338066181) [](ancestors = 338066181)

Reply:
Duh, never mind.  I missed the other classes below.  

---
In reply to: [338198894](https://github.com/Azure/azure-iot-sdk-python/pull/369#discussion_r338198894) [](ancestors = 338198894,338066181)